dfsnew/download.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2015 The Serviced Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dfs
+
+import (
+	"errors"
+
+	"github.com/control-center/serviced/commons"
+	"github.com/control-center/serviced/datastore"
+	"github.com/control-center/serviced/dfsnew/docker"
+	"github.com/control-center/serviced/domain/registry"
+	"github.com/zenoss/glog"
+)
+
+var (
+	ErrImageCollision = errors.New("registry image collision")
```

Comment:
Looking at this from an external perspective, as a consumer of the DFS Interface, would it be better to have all of the package-specific errors defined in the same file as the interface?